src/main/java/net/javadiscord/javabot/listener/PingableNameListener.java

```diff
@@ -0,0 +1,123 @@
+package net.javadiscord.javabot.listener;
+
+import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.TextChannel;
+import net.dv8tion.jda.api.events.guild.member.GuildMemberJoinEvent;
+import net.dv8tion.jda.api.events.guild.member.update.GuildMemberUpdateNicknameEvent;
+import net.dv8tion.jda.api.hooks.ListenerAdapter;
+import net.javadiscord.javabot.Bot;
+import net.javadiscord.javabot.util.GuildUtils;
+import net.javadiscord.javabot.util.StringUtils;
+
+import java.io.IOException;
+import java.net.URL;
+import java.util.*;
+import java.util.stream.Collectors;
+
+/**
+ * Checks User's names & nicknames and makes sure they're pingable.
+ */
+@Slf4j
+public class PingableNameListener extends ListenerAdapter {
+
+	private static final String ADJECTIVES_URL = "https://gist.githubusercontent.com/karlbright/f91229b8c5ac6f4291dc/raw/4a69c2c50b88ee4559b021c443fee899535adc60/adjectives.txt";
+	private static final String NOUNS_URL = "https://raw.githubusercontent.com/hugsy/stuff/main/random-word/english-nouns.txt";
+	private static final Random random = new Random();
+	private final List<String> nouns;
+	private final List<String> adjectives;
+
+	/**
+	 * Constructs a new PingableNameListener and loads nouns & adjectives.
+	 */
+	public PingableNameListener() {
+		nouns = readStrings(NOUNS_URL);
+		adjectives = readStrings(ADJECTIVES_URL);
+		log.info("Loaded {} Nouns!", nouns.size());
+		log.info("Loaded {} Adjectives!", adjectives.size());
+	}
+
+	@Override
+	public void onGuildMemberJoin(GuildMemberJoinEvent event) {
```

Comment:
The methods `onGuildMemberJoin` and `onGuildMemberUpdateNickname` could be merged together.

Reply:
How would that work?

Reply:
What do you mean?

Reply:
```java
	@Override
	public void onGuildMemberJoin(GuildMemberJoinEvent event) {
		checkNickname(event.getMember,null);
	}
	@Override
	public void onGuildMemberUpdateNickname(GuildMemberUpdateNicknameEvent event) {
		checkNickname(event.getMember,event.getNewNickname());
	}
	private void checkNickname(Member member, String name){
		  if (!(name==null||isPingable(name)) && !isPingable(member.getUser().getName()) && !canBypassCheck(member)) {
			changeName(member, Bot.config.get(member.getGuild()).getModeration().getLogChannel());
		}
	}
```